core/src/main/java/com/graphhopper/routing/InstructionsFromEdges.java

```diff
@@ -332,6 +332,9 @@ public void finish() {
     }
 
     private int getTurn(EdgeIteratorState edge, int baseNode, int prevNode, int adjNode, String name, String destinationAndRef) {
+        if (edge.getEdge() == prevEdge.getEdge())
+            // this is the simplest turn to recognize, a plain u-turn.
+            return Instruction.U_TURN_UNKNOWN;
```

Comment:
Basically, this is my entire fix. But since I'm not very familiar with the instruction calculation this might not be the best place to handle the u-turn.

Reply:
@boldtrn, maybe you are more familiar with this?

Reply:
Sorry for the late reply here, this actually seems quite good, no complaints from my side 👍